client/dive-common/use/useModeManager.ts

```diff
@@ -270,14 +277,20 @@ export default function useModeManager({
     };
 
     if (selectedTrackId.value !== null) {
-      const track = trackMap.get(selectedTrackId.value);
+      let track = trackMap.get(selectedTrackId.value);
+      if (selectedCamera.value !== 'default') {
+        track = camTrackMap[selectedCamera.value].get(selectedTrackId.value);
+      }
       if (track) {
         // newDetectionMode is true if there's no keyframe on frameNum
         const { features, interpolate } = track.canInterpolate(frameNum);
         const [real] = features;
 
         // Give each recipe the opportunity to make changes
         recipes.forEach((recipe) => {
+          if (!track) {
+            return;
+          }
```

Comment:
What is this doing?  We're inside an `if (track)` block already.

FWIW, that `if (track)` is also probably wrong because it implies that the else case is a valid execution path when it should probably be a runtime error.

Reply:
it's a typescript oddity that it believe that track can be undefined even though the forEach is inside of an `if (track)`.  I'll see what I can do about it.

Reply:
> it's a typescript oddity

This would go beyond "oddity"--either there's some weird dynamics in this code that we're missing (perhaps due to lexical scoping of the `forEach()` callback?), or there's a typescript bug. What error do you get when you omit the if block? You can also use this technique (https://stackoverflow.com/a/66124742/1886928) to force the compiler to tell you what it thinks the type of `track` is at that point in the code.

Reply:
It believes that within the forEach that track can be `Track | undefined` and doesn't want to use it as an argument to `recipe.update`
![image](https://user-images.githubusercontent.com/61746913/159532932-d7cc0b5b-e2b8-4b64-b867-3d0d9f75b938.png)

Reply:
And how about outside of the forEach callback (3 lines up, on the `track.canInterpolate()` invocation)?

Reply:
It's fine there and is known as a track.  I've updated my last comment and it might have been too late.  But there is an issue exactly like this and has a simplistic demo showing the problem.  I think I've seen this before as well.
https://github.com/microsoft/TypeScript/issues/29916

Reply:
Thanks for looking into this.

I know this is nitpicky, but it happens from time to time.  When typescript misbehaves, I think it would be better to put in a `// @ts-ignore` than to insert code that handles what we know to be an invalid execution path.  That `return` should be unreachable, and I wouldn't blame a reasonable developer for seeing that in the future and ripping it out because it looks like it shouldn't be there.

Reply:
And because you don't want to disable type checks for that whole line, you can probably add something like this:

``` ts
// @ts-ignore
const _track: Track = track;
const changes = recipe.update(eventType, frameNum, _track, [data], key);
```

Reply:
Was implementing and did a quick search for the eslint-disable command and found this:
![image](https://user-images.githubusercontent.com/61746913/159537277-f2544b56-0e3e-4247-8ad0-0fd6a8b8e868.png)
apparently my mind isn't deteriorating as quickly as I thought, we have run into this before.

Reply:
You can also use a type assertion here ("casting" with the `as` operator) to avoid having to disable the ts-lint rules.